### PR TITLE
home-assistant: list outdated libs in parse-requirements.py

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

More closely watch for outdated dependencies  used in the home-assistant ecosystem.

Before #111661 it would print this at the end of `./parse-requirements.py` run:

```
Outdated packages:
 - bt_proximity 0.0.20180217 < 0.2
 - brother 0.1.18 < 0.1.20
 - plexapi 4.1.2 < 4.2.0
 - plexauth 0.0.5 < 0.0.6
 - py-cpuinfo 5.0.0 < 7.0.0
 - fixerio 0.1.1 < 1.0.0a0
 - linode-api 4.1.8b1 < 4.1.9b1
 - libsoundtouch 0.4.0 < 0.8
 - python-telegram-bot 13.0 < 13.1
 - holidays 0.10.3 < 0.10.4
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/home-assistant/parse-requirements.py [hunk 1]
```diff
@@ -156,8 +165,11 @@ def main() -> None:
             # Some requirements are specified by url, e.g. https://example.org/foobar#xyz==1.0.0
             # Therefore, if there's a "#" in the line, only take the part after it
             req = req[req.find("#") + 1 :]
-            name = req.split("==")[0]
+            name, hass_version = req.split("==", maxsplit=1)
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/parse-requirements.py:168]:
```suggestion
            name, required_version = req.split("==", maxsplit=1)
```

Comment (mweinelt) [reply pkgs/servers/home-assistant/parse-requirements.py]: Ah yeah, that is indeed a bit ambiguous.

File: pkgs/servers/home-assistant/parse-requirements.py [hunk 2]
```diff
@@ -189,6 +201,11 @@ def main() -> None:
         f.write("  };\n")
         f.write("}\n")
 
+    if outdated:
+        print("\nOutdated packages:")
+        for package, version in outdated.items():
+            print(f" - {package} {version['nixpkgs']} < {version['hass']}")
```
Comment (Mic92) [inline pkgs/servers/home-assistant/parse-requirements.py:207]:
```suggestion
            print(f" - {package}: current: {version['nixpkgs']} < wanted: {version['hass']}")
```

Comment (mweinelt) [reply pkgs/servers/home-assistant/parse-requirements.py:207]: I think that makes it unnecessarily noisy:

```
Outdated packages:
 - bt_proximity: current: 0.0.20180217 < wanted: 0.2
 - brother: current: 0.1.18 < wanted: 0.1.20
 - plexapi: current: 4.1.2 < wanted: 4.2.0
 - plexauth: current: 0.0.5 < wanted: 0.0.6
 - py-cpuinfo: current: 5.0.0 < wanted: 7.0.0
 - fixerio: current: 0.1.1 < wanted: 1.0.0a0
 - linode-api: current: 4.1.8b1 < wanted: 4.1.9b1
 - libsoundtouch: current: 0.4.0 < wanted: 0.8
 - python-telegram-bot: current: 13.0 < wanted: 13.1
 - holidays: current: 0.10.3 < wanted: 0.10.4
```

Maybe a table would be better.

Comment (mweinelt) [reply pkgs/servers/home-assistant/parse-requirements.py:207]:
```
             Outdated dependencies              
┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━┓
┃ Package             ┃ Current      ┃ Wanted  ┃
┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━┩
│ bt_proximity        │ 0.0.20180217 │ 0.2     │
│ brother             │ 0.1.18       │ 0.1.20  │
│ plexapi             │ 4.1.2        │ 4.2.0   │
│ plexauth            │ 0.0.5        │ 0.0.6   │
│ py-cpuinfo          │ 5.0.0        │ 7.0.0   │
│ fixerio             │ 0.1.1        │ 1.0.0a0 │
│ linode-api          │ 4.1.8b1      │ 4.1.9b1 │
│ libsoundtouch       │ 0.4.0        │ 0.8     │
│ python-telegram-bot │ 13.0         │ 13.1    │
│ holidays            │ 0.10.3       │ 0.10.4  │
└─────────────────────┴──────────────┴─────────┘
```

Comment (Mic92) [reply pkgs/servers/home-assistant/parse-requirements.py]: fancy.

**Pull request merged**